Backend/app/controllers/customer.controller.js

```diff
@@ -1,34 +1,46 @@
-const Customer = require("../models/customer.model.js");
+const Customer = require('../models/customer.model');
+const CaseNote = require('../models/casenote.model');
 
 // Create and Save a new Customer
 
-const getAllUsers = ((req, res) => {
-    Customer.retrieveAll()
-    .then(function(results) {
-        res.send({'customers': results});
-    })
-    .catch(function(err){
-        console.error(err);
-        // TODO error handling
-        res.send({'error': err});
-    });
-});
+const customerController = {
+  async getAllUsers(req, res) {
+    try {
+      const results = await Customer.retrieveAll();
+      res.send({ customers: results });
+    } catch (err) {
+      console.error(err);
+      // TODO error handling
+      res.status(500);
+      res.send({ error: err });
+    }
+  },
+  async getAlertCase(req, res) {
+    try {
+      const { user_id } = req.params;
+      const caseNote = await Customer.getAlertCase(user_id);
+      res.json(caseNote);
+    } catch (err) {
+      console.error(err);
+      // TODO error handling
+      res.status(500);
+      res.send({ error: err });
+    }
+  },
+  async setAlertCase(req, res) {
+    try {
+      const { user_id } = req.params;
+      const { case_id } = req.query;
+      await Customer.setAlertCaseId(user_id, case_id);
+      res.send({ success: true });
+    } catch (err) {
+      console.error(err);
+      // TODO error handling
+      res.status(500);
+      res.send({ error: err });
+    }
+  },
+};
 
-const getCases = ((req, res) => {
-    const user_id = req.query.user_id;
-    const start_date = req.query.start_date;
-    const end_date = req.query.end_date;
-    Customer.getCases(user_id, start_date, end_date)
-    .then(function(results) {
-        res.send({'cases': results});
-    })
-    .catch(function(err){
-        console.error(err);
-        res.send({'error': err});
-    });
-});
```

Comment:
Is this the function @yrao6597 worked on? Don't delete it, but convert to async function.

Reply:
Yes it’s the one I worked on and pushed